test/resources/reftest-analyzer.js

```diff
@@ -214,9 +214,9 @@ window.onload = function () {
         const extra = match[4];
 
         gTestItems.push({
-          pass: !state.match(/FAIL$/),
+          pass: !state.endsWith("FAIL"),
           // only one of the following three should ever be true
-          unexpected: !!state.match(/^TEST-UNEXPECTED/),
+          unexpected: state.startsWith("TEST-UNEXPECTED"),
```

Comment:
Please note that changing the regular expression functions also triggered [this rule](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-string-starts-ends-with.md).